### PR TITLE
upgrade golang and ccs-gm

### PR DESCRIPTION
**Pull request by SamYuan1990**
Signed-off-by: Sam Yuan <yy19902439@126.com>

### CODE REVIEW

File: .github/workflows/CCSPKUsm2encrypt.yml [hunk 1]
```diff
@@ -4,23 +4,10 @@ on:
...
-          go-version: 1.16
-    - run: git clone https://github.com/guanzhi/GmSSL.git
-      name: Clone GMSSL
-    - run: cd GmSSL; SYSTEM=`uname -s` ./config --prefix=/usr/local/gmssl; make; sudo make install_sw  
-      name: Config, make and install GMSSL
-    - run: cd interop; go mod vendor
-    - run: cd interop; go test -v util_test.go sm2Interop_test.go tjSM2.go ccsSM2.go pkuSM2.go type.go
-      name: sm2 data interaction check
\ No newline at end of file
+  call-workflow-1-in-local-repo:
```
Comment (davidkhala) [inline .github/workflows/CCSPKUsm2encrypt.yml:8]: `call-workflow-1-in-local-rep`
改个名字？

File: .github/workflows/sm3.yml [hunk 2]
```diff
@@ -7,16 +7,16 @@ on:
...
     steps:
     - uses: actions/checkout@v2
     - uses: actions/setup-go@main
       with:
-          go-version: 1.16
+          go-version: 1.18
     - run: git clone https://github.com/guanzhi/GmSSL.git
       name: Clone GMSSL
-    - run: cd GmSSL; SYSTEM=`uname -s` ./config --prefix=/usr/local/gmssl; make; sudo make install_sw  
+    - run: cd GmSSL; git checkout master; SYSTEM=`uname -s` ./config --prefix=/usr/local/gmssl; make; sudo make install_sw
```
Comment (davidkhala) [inline .github/workflows/sm3.yml:19]: 这里可以用multiple checkout 法， 见
https://github.com/Hyperledger-TWGC/pku-gm/blob/428b668e73b12e3319472945c951b01fdcad2dc5/.github/workflows/ci.yml#L10

Comment (SamYuan1990) [reply .github/workflows/sm3.yml]: let's see if checkout master branch or default branch. it seems `https://github.com/guanzhi/GmSSL.git` fail build with default branch for now with CI.

Comment (SamYuan1990) [reply .github/workflows/sm3.yml]: let's see if checkout master branch or default branch. it seems `https://github.com/guanzhi/GmSSL.git` fail build with default branch for now with CI.

**Review by davidkhala — APPROVED** [review]
LGTM

**Pull request merged**